From 73baceed3ad79bf1bb9885dbac4294375fc90a75 Mon Sep 17 00:00:00 2001
From: allen yan <yanwei@marvell.com>
Date: Wed, 11 Jan 2017 23:10:26 +0800
Subject: [PATCH 0737/1345] uart: a3700: add port soft reset to hardware
 initialization

commit  fa5a04a04b3d3d32df6f2f6f06579f89d76b8e74 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- The existing uart driver relied on the bootloader
  to initialize the port. However, the secondary uart
  port may not be initialized properly in early boot
  stage.
- This patch added the uart soft reset in probe()
  function, which works for all the ports.

Change-Id: I995b62520007a15417c3325df977af32ac0d786f
Signed-off-by: allen yan <yanwei@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35470
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/mvebu-uart.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/tty/serial/mvebu-uart.c b/drivers/tty/serial/mvebu-uart.c
index 8f20a14..1222e9f 100644
--- a/drivers/tty/serial/mvebu-uart.c
+++ b/drivers/tty/serial/mvebu-uart.c
@@ -795,6 +795,11 @@ static int mvebu_uart_probe(struct platform_device *pdev)
 	port->private_data = data;
 	platform_set_drvdata(pdev, data);
 
+	/* UART Soft Reset*/
+	writel(CTRL_SOFT_RST, port->membase + REG_CTRL(data));
+	udelay(1);
+	writel(0, port->membase + REG_CTRL(data));
+
 	ret = uart_add_one_port(&mvebu_uart_driver, port);
 	if (ret)
 		return ret;
-- 
1.7.9.5

